cmd/capture.go

```diff
@@ -35,13 +40,29 @@ by passing the function name symbol and the binary args.
 `,
 	Example: "  harpoon -f main.doSomething ./command arg1 arg2 ...",
 	Run: func(cmd *cobra.Command, args []string) {
-		opts := captor.CaptureOptions{
+
+		functionSymbolList := strings.Split(functionSymbols, ",")
+
+		captureOpts := captor.CaptureOptions{
 			CommandOutput: commandOutput,
 			LibbpfOutput:  libbpfOutput,
-			Save:          save,
-			Directory:     directory,
 		}
-		captor.Capture(functionSymbols, args, opts)
+		for _, functionSymbol := range functionSymbolList {
+			syscalls, err := captor.Capture(functionSymbol, args, captureOpts)
+			if err != nil {
+				fmt.Printf("error capturing syscall: %v", err)
+				os.Exit(1)
```

Comment:
This is a bad pattern.

No os.Exit should be found outside main.go

It's outside the scope of your PR, and could be addressed in another one.

But here I would use RunE not Run because RunE signature supports returning an error.

So the os.Exit can be found in main.go only

This way you can test your code, and you will have one code to handle things

Reply:
Please note it could also mean you could use errors wrapping

So instead of this
                                fmt.Printf("error capturing syscall: %v", err)

You would use return fmt.Errorf("error capturing syscall: %w", err)

This way the fmt.Printis only present in main.go

Reply:
I'm removing this bad pattern from `capture.go`, but for the other files I'll open another PR.
I don't want to fill this PR with too many commits out of scope.